### PR TITLE
Feature/rabbitmq queue

### PR DESCRIPTION
**Pull request by h4ck3rm1k3**
superseding https://github.com/saltstack/salt/pull/26286

### CODE REVIEW

File: salt/modules/rabbitmq.py [hunk 1]
```diff
@@ -578,6 +579,29 @@ def list_queues_vhost(vhost, runas=None, *kwargs):
     return res
 
 
+def queue_vhost_exists(queue, vhost, runas=None, *kwargs):
```
Comment (cachedout) [inline salt/modules/rabbitmq.py]: What's the purpose of putting kwargs into a tuple of positional arguments? I don't see a need for this.

File: salt/modules/rabbitmqadmin.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    durable = translate_boolean(durable)
+    auto_delete = translate_boolean(auto_delete)
+
+    if runas is None:
+        runas = salt.utils.get_user()
+    res = __salt__['cmd.run']('rabbitmqadmin declare queue --vhost={0} name={1} durable={2} auto_delete={3}'.format(
+        vhost, name, durable, auto_delete),
+                              python_shell=False, runas=runas)
+    log.debug(res)
```
Comment (cachedout) [inline salt/modules/rabbitmqadmin.py]: It's going to be extremely difficult to discern what these log messages are or where they are coming from. Please either change these to include info about their origin or take them out.

File: salt/modules/rabbitmqadmin.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+    '''
+
+    if runas is None:
+        runas = salt.utils.get_user()
+    res = __salt__['cmd.run']('rabbitmqadmin declare binding --vhost={0} source={1} destination={2} destination_type={3} routing_key={4}'.format(vhost, source, destination, destination_type, routing_key),
+                              python_shell=False,
+                              runas=runas)
+    log.debug(res)
+    msg = 'Declared'
```
Comment (cachedout) [inline salt/modules/rabbitmqadmin.py]: We don't need this assignment. It can just be passed in as an arg on the following line.

**Comment by cachedout** [comment]
@h4ck3rm1k3 Please take a look at the comments I left inline and then we'll get this merged. Thanks.

**Comment by cachedout** [comment]
Ping @h4ck3rm1k3 Did you get a chance to review the above changes?

**Comment by h4ck3rm1k3** [comment]
I have not, we have moved back east and this is the first time I am able to get online. I hope this weekend.

**Comment by h4ck3rm1k3** [comment]
I have not had time, give me another week.

**Comment by cachedout** [comment]
@h4ck3rm1k3 No problem. I'll close this for now and when you're ready, you can just re-open it and we'll get it merged. Thanks!

**Pull request closed without merging**